zedd-app/src/components/AppBody.tsx

```diff
@@ -247,7 +251,22 @@ export const AppBody = observer(
             </Button>
             <ButtonGroup variant='outlined'>
               <Tooltip title='Fill currently shown empty days with ERSATZ task.' arrow>
-                <Button size='large' onClick={(_) => state.fillErsatz(state.showing)}>
+                <Button
+                  size='large'
+                  onClick={(_) => {
+                    if (settings.location) {
+                      getHolidays(
+                        (state.showing.start as Date).getFullYear(),
+                        settings.location!.code,
+                        settings.federalState?.code,
+                      ).then((holidays) => {
+                        state.fillErsatz(state.showing, holidays)
+                      })
+                    } else {
+                      state.fillErsatz(state.showing, [])
+                    }
```

Comment:
please use async/await and add error handling (you can simulate errors by adding a typo to the api url.